lib/urlwatch/reporters.py

```diff
@@ -978,3 +978,65 @@ def submit(self):
 
         for chunk in chunkstring(text, self.MAX_LENGTH, numbering=True):
             asyncio.run(xmpp.send(chunk))
+
+class ProwlReport(TextReporter):
+    """Send a detailed notification via prowlapp.com"""
+
+    __kind__ = 'prowl'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def submit(self):
+        api_base = 'https://api.prowlapp.com/publicapi/'
+        api = {
+            'add': api_base + 'add'
+        }
+
+        text = '\n'.join(super().submit())
+
+        if not text:
+            logger.debug('Not calling Prowl API (no changes)')
+            return
+
+        filtered_job_states = list(self.report.get_filtered_job_states(self.job_states))
+        subject_args = {
+            'count': len(filtered_job_states),
+            'jobs': ', '.join(job_state.job.pretty_name() for job_state in filtered_job_states),
+        }
+
+        # 'subject' used in the config file, but the API
+        # uses what might be called the subject as the 'event'
+        event = self.config['subject'].format(**subject_args)
+
+        # 'application' is prepended to the message in prowl,
+        # to show the source of the notification. this too,
+        # is user configurable, and may reference subject args
+        application = self.config['application'].format(**subject_args)
```

Comment:
Use this instead, so that the user can leave out the "application" config key:

```
    application = self.config.get('application')
    if application is not None:
        application = application.format(**subject_args)
    else:
        application = '{0} v{1}'.format(urlwatch.pkgname, urlwatch.__version__)
```